### PR TITLE
Adding `ard_complex()`

### LINKED ISSUE #103: Add experimental badge to `as_nested_list()` function
I need to spend more time investigating the structure of JSON objects, so let's mark this function as experimental before a release is made.

Since lifecycle is a build-time dependency, we don't need to add it to the DESCRIPTION file.

```r
lifecycle::badge("experimental")
#> "\\ifelse{html}{\\href{https://lifecycle.r-lib.org/articles/stages.html#experimental}{\\figure{lifecycle-experimental.svg}{options: alt='[Experimental]'}}}{\\strong{[Experimental]}}"
```

### LINKED ISSUE #108: We need a more generalized ARD function....
I think something like this would work:

```r
ard_custom()
```

That would have the same API `ard_continuous()`, expect the functions passed in the `ard_continuous(statistic)` argument would accept more arguments than just the subsetted vector. I think if the functions accepted the following, it would be flexible enough to do what we need
- `x` the subsetted primary vector being analyzed
- `data` the subsetted (`by`/`strata`) data frame
- `data_full` the full data frame
- `variable` the string of the variable being analyzed
- `by` character vector of the `by` variables
- `strata` character vector of the `strata` variables
- `...` all functions should accept `...` so this can be extended in the future

The reason this comes to mind is that I was working on proportion confidence intervals, and some of the more complex calculations have a 2nd stratifying variable. Also, we dealt with this in `gtsummary::tbl_custom_summary()` because it's a situation that just comes up every so often.

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
- Added `ard_complex()` that can handle summaries similar to `ard_continuous()`, but it not limited to univariable summaries. I marked this function as experimental because I will need this function to make `gtsummary::tbl_custom_summary()` work, and would prefer to leave it marked as experimental until that integration is complete just in case I need to update the function to make it fully compatible.
- Added lifecycle badges
- Renamed `max_values()` to `maximum_variable_values()`
- Documenting some un-exported functions that I had missed earlier

**Reference GitHub issue associated with pull request.** _e.g., 'closes #1'_
closes #108
closes #103 

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] Ensure all package dependencies are installed: `devtools::install_dev_deps()`
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")`
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

**Comment by ddsjoberg** [comment]
Hi @bzkrouse ! This one is ready for review, thanks!

**Review by bzkrouse — APPROVED** [review]
based on a very brief review!

File: R/ard_complex.R [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                        stat_labels = everything() ~ default_stat_labels()) {
+  # check inputs ---------------------------------------------------------------
+  check_not_missing(data)
+  check_not_missing(variables)
+  check_not_missing(statistics)
+  check_class_data_frame(data = data)
+  check_class(class = c("list", "formula"), statistics = statistics, allow_null = FALSE)
+
+  # process inputs -------------------------------------------------------------
+  process_selectors(data, variables = variables)
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/ard_complex.R:64]: Ahh, good point! I also updated this a few lines below: (previously was using the curly embrace)

```r
ard_continuous(
    data = data,
    variables = all_of(variables)
```

**Comment by ddsjoberg** [comment]
@bzkrouse thank you for the review!!!

**Pull request merged**